### PR TITLE
Replace react warnings with errors

### PR DESCRIPTION
**Pull request by FK78**
**Overall change:** _React warnings have been replaced with react errors._

**Code changes:**

- _Developers can't push their changes until react errors are resolved._
- _Unused variables have been removed._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by dr3 — APPROVED** [review]

File: src/testHelpers/setupTests.js [hunk 1]
```diff
@@ -12,9 +12,6 @@ const REACT_ERRORS_REGEX = new RegExp(REACT_ERRORS.join('|'));
 
 const { error } = console;
 
-let reactErrorsCountPerTestSuite = 0;
-const reactErrorsLimitPerTestSuite = 6; // The goal is to have zero React errors per suite so keep fixing errors and lowering this limit
-
 const didSuppressWarning = message => {
```
Comment (jamesbhobbs) [inline src/testHelpers/setupTests.js:15]: I don't really follow what this is doing or why it's needed. can you explain?

Comment (jamesbhobbs) [reply src/testHelpers/setupTests.js:15]: Suppresses warnings when we want to write tests to violate prop types, [an example](https://github.com/bbc/simorgh/blob/latest/src/app/containers/Image/index.test.jsx#L12).

**Review by jamesbhobbs — APPROVED** [review]

**Comment by dr3** [comment]
For the benefit of those not following on slack :)

>Benjamin Hobbs:house_with_garden: 11:28 AM
>Question on the above
>@ Fahad @ drew is that whole file just now erroring if count of prop type wanring is higher than number set, and now we're changing it to error if it's higher than 0? (edited) 
>
>drew:spotify: 11:31 AM
>Its moving from 'You can cause 6 prop type warnings before the test suite fails' to 'You cant throw prop type warnings in the tests'
>
>Benjamin Hobbs:house_with_garden: 11:31 AM
>there's a npm dep for exactly that, i don't see why we'd write and maintain custom code
vhttps://www.npmjs.com/package/jest-prop-type-error
>
>drew:spotify: 11:32 AM
>TIL
>after a 30 second scan that doesnt allow you to supress prop type warnings though
ours supports that
>
>Benjamin Hobbs:house_with_garden: 11:33 AM
>where and why do we do that?
>
>drew:spotify: 11:34 AM
>https://github.com/bbc/simorgh/blob/latest/src/app/containers/Image/index.test.jsx#L12 (edited) 
>Tests where you are required to violate proptypes, to write the test
>
>Benjamin Hobbs:house_with_garden: 11:37 AM
>ok then, that makes sense

File: src/testHelpers/setupTests.js (hunk 1 above)
Comment (dr3) [reply src/testHelpers/setupTests.js:15]: https://github.com/bbc/simorgh/pull/3525#issuecomment-527404107

**Pull request merged**